.github/workflows/publish.yml

```diff
@@ -14,52 +14,128 @@ on:
         default: ''
         type: string
       pure_python_wheel:
-        description: Whether to build a pure Python wheel - if not, separate wheels are built for each platform.
+        description: Whether to build a pure Python wheel - if not, separate wheels are built for each platform
         required: false
         default: false
         type: boolean
+      platforms:
+        description: List of target platforms
+        required: false
+        default: 'ubuntu-20.04 windows-2019 macos-10.15'
+        type: string
+      upload_to_pypi:
+        description: Always upload to PyPI - if not, only upload tags
+        required: false
+        default: false
+        type: boolean
+      repository_url:
+        description: The PyPI repository URL to use
+        required: false
+        default: ''
+        type: string
+      submodules:
+        description: Whether to checkout submodules
+        required: false
+        default: true
+        type: boolean
+    secrets:
+      pypi_password:
+        required: false
 
 jobs:
 
+  generate_platform_matrix:
+    name: Generate build_wheels matrix
+    runs-on: ubuntu-latest
+    if: ${{ !inputs.pure_python_wheel }}
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - id: set-matrix
+        run: |
+          FILTER='split(" ") as $platforms | {"os": $platforms}'
+          echo "::set-output name=matrix::$( echo ${{ inputs.platforms }} | jq -R -c "$FILTER" )"
```

Comment:
Oh well!

Reply:
I've now found a much better way of doing this by using a Python script to parse a YAML string and return a build matrix as a JSON string! Was experimenting [here](https://github.com/ConorMacBride/test-workflows) and will update this PR now. Basically, it recreates the ["targets:" list](https://openastronomy-azure-pipelines.readthedocs.io/en/latest/publish.html#usage-and-options) from the old template exactly.